pkg/controllermanager/apis/config/validation/validation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	metav1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateControllerManagerConfiguration validates the given `ControllerManagerConfiguration`.
+func ValidateControllerManagerConfiguration(conf *config.ControllerManagerConfiguration) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateControllerManagerControllerConfiguration(conf.Controllers, field.NewPath("controllers"))...)
+	return allErrs
+}
+
+func validateControllerManagerControllerConfiguration(conf config.ControllerManagerControllerConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	projectFldPath := fldPath.Child("project")
+	if conf.Project != nil {
+		allErrs = append(allErrs, validateProjectControllerConfiguration(conf.Project, projectFldPath)...)
+	} else {
+		allErrs = append(allErrs, field.Required(projectFldPath, "must provide a project controller configuration"))
```

Comment:
Why is it required?

Reply:
I added it as required because `.Project` is used at some places w/o any `nil` check. It's anyway defaulted, so this check is also redundant. I'll remove it.